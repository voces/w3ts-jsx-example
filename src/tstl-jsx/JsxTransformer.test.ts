import * as tstl from "typescript-to-lua";
import { readFileSync } from "fs";

it.skip("works", async () => {
	const result = tstl.transpileProject(
		"src/tstl-jsx/JsxTransformer.tsconfig.json",
	);

	expect(result.diagnostics).toHaveLength(0);

	const lua = readFileSync(
		"src/tstl-jsx/__lua__/src/JsxTransformer.sample.lua",
		"utf-8",
	);

	expect(lua).toMatchSnapshot();
});
