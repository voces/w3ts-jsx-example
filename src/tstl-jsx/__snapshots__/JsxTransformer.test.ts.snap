// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"--[[ Generated with https://github.com/TypeScriptToLua/TypeScriptToLua ]]
App = function() return React.createElement(
    Foo,
    {foo = \\"foo\\"},
    {
        React.createElement(
            Bar,
            {bar = 3},
            {
                React.createElement(Baz, {baz = {foo = {\\"value1\\", \\"value2\\"}}}),
                React.createElement(Baz, {})
            }
        )
    }
) end
"
`;
